### PR TITLE
[GRPO]: Fix Entropy Mask Threshold Calculation when using Multi-GPU training

### PR DESCRIPTION
**Pull request by pramodith**
# What does this PR do?
The entropy threshold calculation currently calculates the entropy threshold on local entropies, however this threshold will be inaccurate when training is done on multiple gpus. This PR fixes the bug by using an all gather operation on the entropies before identifying the threshold.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [X] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: trl/trainer/grpo_trainer.py [hunk 1]
```diff
@@ -1068,6 +1042,41 @@ def _get_last_hidden_state(
...
+            `torch.Tensor`:
+                Boolean mask of shape (batch_size, seq_len), where `True` indicates tokens with entropy >= threshold and
+                `False` otherwise.
+        """
+        non_pad_entropies = entropies[mask.bool()].float()
+        if non_pad_entropies.numel() == 0:
+            return torch.zeros_like(entropies, dtype=torch.bool)
+
+        # For multi-GPU training, gather entropies from all processes to compute global quantile
+        if self.accelerator is not None and self.accelerator.num_processes > 1:
```
Comment (LeonEricsson) [inline trl/trainer/grpo_trainer.py]: propose we skip these checks and simplify the code since `self.accelerator` is always initialized and `accelerator.gather()` already handles the single-process case by returning the same tensor

**Review by LeonEricsson — COMMENTED** [review]
looks great, small comment

File: trl/trainer/grpo_trainer.py (hunk 1 above)
Comment (pramodith) [reply trl/trainer/grpo_trainer.py]: Gotcha!

File: trl/trainer/grpo_trainer.py [hunk 2]
```diff
@@ -1068,6 +1042,36 @@ def _get_last_hidden_state(
...
+
+        Returns:
+            `torch.Tensor`:
+                Boolean mask of shape (batch_size, seq_len), where `True` indicates tokens with entropy >= threshold and
+                `False` otherwise.
+        """
+        non_pad_entropies = entropies[mask.bool()].float()
+        if non_pad_entropies.numel() == 0:
+            return torch.zeros_like(entropies, dtype=torch.bool)
+
```
Comment (kashif) [inline trl/trainer/grpo_trainer.py]:
```suggestion

```

Comment (pramodith) [reply trl/trainer/grpo_trainer.py]: Done! Ran precommit to make sure linting is as expected.

**Comment by LeonEricsson** [comment]
Test fails are not from this PR
